File: 05-Navigation-Bar/05-Navigation-Bar/ViewController.m
//
//  ViewController.m
//  05-Navigation-Bar
//
//  Created by meloair on 2018/9/11.
//  Copyright © 2018年 com. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property(nonatomic, strong) UIButton *button;
@end

@implementation ViewController

- (UIButton *) button {
    if (!_button) {
        _button = [UIButton buttonWithType:UIButtonTypeRoundedRect];
        [_button setFrame:CGRectMake((self.view.frame.size.width - 200)/2.0, 60, 200, 40)];
        // sets title for the button
        [_button setTitle:@"Navigate To Next View" forState:
         UIControlStateNormal];
        [_button addTarget:self action:@selector(onTouchHandler:)
          forControlEvents:UIControlEventTouchUpInside];
    }
    return _button;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    // @@ Code Here @@
    [self addButton];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)addButton {;
    [self.view addSubview:self.button];
}


-(IBAction)onTouchHandler:(id)sender{
    ViewController *vc = [[ViewController alloc] init];
    [self.navigationController pushViewController:vc animated:YES];
}

@end

File: S05-Navigation-Bar/S05-Navigation-Bar/ViewController.swift
//
//  ViewController.swift
//  S05-Navigation-Bar
//
//  Created by meloair on 2018/9/17.
//  Copyright © 2018年 com. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    lazy var button : UIButton = {
        let object = UIButton(type: .roundedRect) as UIButton
        object.frame = CGRect(x:(self.view.frame.size.width - 200)/2.0, y:60, width:200, height:40)
        object.setTitle("Navigate To Next View", for: .normal)
        object.addTarget(self, action: #selector(onTouchHandler), for: .touchDown)
        
        return object
    }()

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.view.addSubview(self.button)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @objc func onTouchHandler() {
        let nvc = NextViewController()
        self.navigationController?.pushViewController(nvc, animated: true)
        //self.navigationController?.pushViewController(ViewController(), animated: true)
    }

}

